### ISSUE TITLE
Incorrect hue displayed for hued spell effects on Outlands

### ISSUE THREAD

**Issue by roxya** [ClassicUO/ClassicUO#757]
These images are from beta release:
![image](https://user-images.githubusercontent.com/7057924/66385453-6b011c80-e9b8-11e9-87dc-bc2f67f9abcb.png)
![image](https://user-images.githubusercontent.com/7057924/66385460-6dfc0d00-e9b8-11e9-9d3f-b4f3f553f163.png)
(to me the wrong one looks exactly like hue 1787)

This one from dev preview:
![image](https://user-images.githubusercontent.com/7057924/66385464-705e6700-e9b8-11e9-8377-6a3187518795.png)

**Comment by Djeryv** [comment]
I am going to add to this because it just isn't Outlands.

Most things on the server side do the simple equation of Hue == Hex for that color so assigning colors is mostly simple. Spell effects, and other particle style effects, do not work this way on the server side and original client. Whenever we devs need to assign a color to an effect, we must find the color we want and then subtract "1" from it so it appears properly in the game. It works the same way for sounds in most cases...where we find a sound we want to use and sometimes have to subtract "1" from that as well.

![image](https://user-images.githubusercontent.com/49735284/66636702-577fcc80-ebd7-11e9-8534-064060c5d825.png)

So if I want to have a yellow hue for this spell effect, I would need to set the hue value to 0x5B6.

**Comment by Djeryv** [comment]
To put the above in simpler terms, the original client wants to render effect colors at +1 the hue value sent by the server.

**Comment by andreakarasho** [comment]
Is this true for each effect?

**Comment by Djeryv** [comment]
Fair question. How about I run tests on various different effect styles first and then get back to you. It is best to be sure because it would be annoying to fix one type of effect and break another. Give me some time this morning to do that.

**Comment by Djeryv** [comment]
All of the effect types I tested do the hues the same...where the client wants to add 1 to the hue value sent by the server as I previously described. Now I think it gets tricky with HUE 0 because that actually works fine with the client having no color to the effect...instead of the PURE BLACK of HUE 1. So again, if I set the particle hue to zero in the server script...the client will not color the effect.

ServerHueZero = ClientHueZero
ServerHueOne = ClientHueTwo
ServerHueTwo = ClientHueThree
...and so on...

For RunUO/ServUO scripts it is these types of functions that send the particle effects...just as an FYI:

SendLocationParticles(EffectItem.Create())
SendLocationEffect
SendMovingParticles
SendTargetParticles
SendMovingEffect

**Comment by andreakarasho** [comment]
pushed a fix. News?

**Comment by Djeryv** [comment]
I tried green and purple...and things look good. If I don't set a hue...it is 0 hue as intended for flamestrike...but fire field is black hue when set to zero. For some reason that spell effect is set to "1" when cast.

![image](https://user-images.githubusercontent.com/49735284/67626955-01d33300-f819-11e9-83b1-a67fd6978428.png)

**Comment by Djeryv** [comment]
Here is me casting two different spells where flamestrike is good and fire wall is not.

![image](https://user-images.githubusercontent.com/49735284/67626983-61314300-f819-11e9-9aa6-e1850aa864d9.png)

**Comment by Djeryv** [comment]
I tested again in case it was self corrected. The blue is hue 2 I set for spell effects. The blacks are different hues set like -1, 0, and 1...where zero should be no color at all (meaning the default orange fire).

![image](https://user-images.githubusercontent.com/49735284/70572818-33545380-1b66-11ea-9dcb-e34ba03fbcc8.png)

**Comment by Djeryv** [comment]
I figured out that field spells and spell effects are actually different since the field spells need to make items where spell effects just do an "effect". So after scrutinizing my spell hue scripts (unique to my game) I realized that the last fix Karasho did seems to have actually fixed the problems for me in this regard. So unless others can speak to the contrary (looking at you Outlands players since they have spell hue options)...I think this could be closed.
